daft/runners/ray_runner.py

```diff
@@ -435,14 +434,18 @@ def run_plan(
         plan_scheduler: PhysicalPlanScheduler,
         psets: dict[str, ray.ObjectRef],
         result_uuid: str,
+        results_buffer_size: int | None = None,
     ) -> None:
+        self.results_by_df[result_uuid] = Queue(maxsize=results_buffer_size or -1)
+
         t = threading.Thread(
             target=self._run_plan,
             name=result_uuid,
             kwargs={
                 "plan_scheduler": plan_scheduler,
                 "psets": psets,
                 "result_uuid": result_uuid,
+                "results_buffer_size": results_buffer_size,
```

Comment:
why does the thread need `results_buffer_size` if the queue would already be limited?